src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -372,6 +374,33 @@ public static function underscore($id)
         return strtolower(preg_replace(array('/([A-Z]+)([A-Z][a-z])/', '/([a-z\d])([A-Z])/'), array('\\1_\\2', '\\1_\\2'), str_replace('_', '.', $id)));
     }
 
+    /**
+     * Fetches a variable from the environment.
+     *
+     * @param string The name of the environment variable
+     *
+     * @return scalar The value to use for the provided environment variable name
+     *
+     * @throws EnvNotFoundException When the environment variable is not found and has no default value
+     */
+    protected function getEnv($name)
+    {
+        if (isset($this->envCache[$name]) || array_key_exists($name, $this->envCache)) {
+            return $this->envCache[$name];
+        }
+        if (isset($_ENV[$name])) {
+            return $this->envCache[$name] = $_ENV[$name];
+        }
+        if (false !== $env = getenv($name)) {
+            return $this->envCache[$name] = $env;
+        }
+        if (!$this->hasParameter("env($name)")) {
```

Comment:
Because this method can be called in `getDefaultParameters` during `__construct` of the container (before `$this->parameters` have been set), you need to initialize `$this->parameters` to an array before calling `getDefaultParameters`, otherwise you'll get the following error *if there is an unset environment variable*:

```
                                                                           
  [Symfony\Component\Debug\Exception\ContextErrorException]                
  Warning: array_key_exists() expects parameter 2 to be array, null given  
                                                                           

Exception trace:
 () at /Users/magnus/Developer/Fervo/sftest5/var/cache/dev/appDevDebugProjectContainer.php:3723
 appDevDebugProjectContainer->hasParameter() at /Users/magnus/Developer/Fervo/sftest5/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:397
 Symfony\Component\DependencyInjection\Container->getEnv() at /Users/magnus/Developer/Fervo/sftest5/var/cache/dev/appDevDebugProjectContainer.php:3789
 appDevDebugProjectContainer->getDefaultParameters() at /Users/magnus/Developer/Fervo/sftest5/var/cache/dev/appDevDebugProjectContainer.php:30
 appDevDebugProjectContainer->__construct() at /Users/magnus/Developer/Fervo/sftest5/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:492
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at /Users/magnus/Developer/Fervo/sftest5/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:116
 Symfony\Component\HttpKernel\Kernel->boot() at /Users/magnus/Developer/Fervo/sftest5/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:68
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /Users/magnus/Developer/Fervo/sftest5/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:118
 Symfony\Component\Console\Application->run() at /Users/magnus/Developer/Fervo/sftest5/bin/console:29
```